staging/src/k8s.io/apiserver/plugin/pkg/audit/dynamic/dynamic_test.go

```diff
@@ -233,20 +233,15 @@ func TestDynamic(t *testing.T) {
 		// if the stop signal is not propagated correctly the buffers will not
 		// close down gracefully, and the shutdown method will hang causing
 		// the test will timeout.
-		timeoutChan := make(chan struct{})
 		successChan := make(chan struct{})
-		go func() {
-			time.Sleep(1 * time.Second)
-			timeoutChan <- struct{}{}
-		}()
 		go func() {
 			close(stopChan)
 			d.Shutdown()
 			successChan <- struct{}{}
 		}()
 		for {
```

Comment:
what does the for loop do if both paths return from the Run func()?

Reply:
@MHBauer yes, the for loop is unnecessary. I think we can remove it.